### PR TITLE
Adding make deploy-image target

### PR DESCRIPTION
**Pull request by ewolinetz**

### CODE REVIEW

File: hack/common [hunk 1]
```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+if [ -n "${DEBUG:-}" ]; then
+    set -x
+fi
+
```
Comment (periklis) [inline hack/common:5]: Drop xtrace versbosity.

File: hack/common [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    docker login -u "$username" -p "$token" "$1" > /dev/null
+}
+
+function push_image() {
+    skopeo copy --dest-tls-verify=false docker-daemon:"$1" docker://"$2"
+}
+
+if [ $REMOTE_REGISTRY = false ] ; then
+    : # skip
+else
```
Comment (periklis) [inline hack/common:60]: Drop unused code branch

File: hack/deploy-image.sh [hunk 3]
```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+if [ "${DEBUG:-}" = "true" ]; then
+  set -x
+fi
```
Comment (periklis) [inline hack/deploy-image.sh:5]: Drop xtrace verbosity.

File: hack/common (hunk 1 above)
Comment (ewolinetz) [reply hack/common:5]: why would we drop the ability to set DEBUG?

Comment (periklis) [reply hack/common:5]: xtrace is not debug per se. It obscure test/execution logs and considered harmful if by mistake password etc are exposed. You can always set `bash -x` in front of your script. We have already started to drop it from our repos, e.g. openshift/luster-logging-operator#383 and openshift/origin-aggregated-logging#1832

File: hack/common [hunk 4]
```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+alias oc=${OC:-oc}
+
+repo_dir="$( cd "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P )/.."
+
+IMAGE_TAG=${IMAGE_TAG:-"quay.io/openshift/origin-elasticsearch-proxy:latest"}
+
+function login_to_registry() {
```
Comment (jcantrill) [inline hack/common:8]: I  have always questioned the need for any of this complicated logic.  This is a dev operation.  Is there any reason we simply can not assume the user running the target can also push to the registry in order to simplify the logic.  Can we just not fail with a simple message that "Are you certain you can log into XXXXX". My preference would be to login simply with something like`oc login -u $(oc whoami) -p $(oc whoami -t)`

File: hack/common [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        fi
+    fi
+    docker login -u "$username" -p "$token" "$1" > /dev/null
+}
+
+function push_image() {
+    skopeo copy --dest-tls-verify=false docker-daemon:"$1" docker://"$2"
+}
+
+if [ ${REMOTE_REGISTRY:-true} = true ] ; then
```
Comment (jcantrill) [inline hack/common:52]: AFAIK the registry is always "remote" .  You can use the same logic to "push" to a local setup though it just takes more time. Given we have few local setups these days I think we can make some opinionated assumptions.

Comment (ewolinetz) [reply hack/common:52]: ack

File: hack/common (hunk 4 above)
Comment (ewolinetz) [reply hack/common:8]: what if i'm running locally as `system:admin`, i'll be required to log in as `kubeadmin` then when i want to push an image? i currently like using the `PUSH_USER` and `PUSH_PASSWORD` mechanic

Comment (jcantrill) [reply hack/common:8]: This seems rare IMO and I have seen problems myself when developing locally where the context is not being reset correctly

**Pull request merged**